tiledb/api/src/query/conditions.rs

```diff
@@ -714,4 +848,39 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn display() {
+        let qc_cmp = QC::field("field").lt(5);
+        assert_eq!("field < 5", qc_cmp.to_string());
+
+        let qc_setmemb =
+            QC::field("field").is_in(["one", "two", "three"].as_slice());
+        assert_eq!("field IN (one, two, three)", qc_setmemb.to_string());
+
+        let qc_nullness = QC::field("field").not_null();
+        assert_eq!("field IS NOT NULL", qc_nullness.to_string());
+
+        let qc_comb = qc_cmp.clone() & qc_setmemb.clone();
+        assert_eq!(
+            format!("({}) AND ({})", qc_cmp, qc_setmemb),
+            qc_comb.to_string()
+        );
+
+        let qc_neg = !qc_nullness.clone();
+        assert_eq!(format!("NOT ({})", qc_nullness), qc_neg.to_string());
+
+        /* parentheses should leave no ambiguity */
+        let atom = QC::field("x").lt(5);
+        let qc_tree = (atom.clone() | atom.clone())
+            & (!atom.clone() | !(atom.clone() & atom.clone()));
+
+        assert_eq!(
+            "((x < 5) OR (x < 5)) AND ((NOT (x < 5)) OR (NOT ((x < 5) AND (x < 5))))",
+            qc_tree.to_string()
+        );
+    }
+
+    #[test]
+    fn display_ambiguity() {}
```

Comment:
Not sure I understand what this test is testing.

Reply:
That seems consistent with the name of the test!


To be serious, I started putting the nested case here but then decided to just have the one test function